shared/src/main/scala/ListHelpers.scala

```diff
@@ -67,6 +103,54 @@ object ListHelpers:
     moldHelper(content, shape, 0)
   end mold
 
+  def sortBy(iterable: VList, key: VFun)(using ctx: Context): VList =
+    val branches = key.originalAST match
+      case Some(lam) => lam.body
+      case None =>
+        return VList(
+          iterable.zipWithIndex
+            .sorted((a, b) =>
+              MiscHelpers.compareExact(
+                key.executeResult(a(0), a(1), List(a(0))),
+                key.executeResult(b(0), b(1), List(b(0)))
+              )
+            )
+            .map(_._1)*
+        )
+
+    if branches.length < 2 then
+      return VList(
+        iterable.zipWithIndex
+          .sorted((a, b) =>
+            MiscHelpers.compareExact(
+              key.executeResult(a(0), a(1), List(a(0))),
+              key.executeResult(b(0), b(1), List(b(0)))
+            )
+          )
+          .map(_._1)*
+      )
+
+    val out = iterable.zipWithIndex
+      .sortWith { (a, b) =>
+        val (aRes, bRes) =
+          branches.view
+            .map { branch =>
+              val f = VFun.fromLambda(AST.Lambda(1, List.empty, List(branch)))
+              (
+                f.execute(a(0), a(1), List(a(0))),
+                f.execute(b(0), b(1), List(b(0)))
+              )
+            }
+            .find(_ != _)
+            .getOrElse((ctx.settings.defaultValue, ctx.settings.defaultValue))
+        MiscHelpers.compareExact(aRes, bRes) < 0
```

Comment:
```suggestion
            .map { case (aRes, bRes) => MiscHelpers.compareExact(aRes, bRes) < 0 }
            .getOrElse(false)
```
On second thought, there's no need to use the default value because all we want to do is return false in the case that they're equal.

(this suggestion is part of the review but it's not a change I 100% want to happen, it's just a suggestion you can reject)

Reply:
This seems to error anyway, saying it's a `Unit`, not a `Boolean`

Reply:
See the new suggestion below